src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,11 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .filter(optionalUser -> optionalUser.getAge() != null)
+                .filter(optionalUser -> optionalUser.getAge() >= 30)
+                .filter(optionalUser -> optionalUser.getAge() <= 45)
```

Comment:
filter 내부에서 중복으로 사용하는 getter 호출을 다음과 같이 제거할 수 있을 것 같아요 :)
```suggestion
        return Optional.ofNullable(user)
                .map(User::getAge)
                .filter(age -> age != null)
                .filter(age -> age >= 30)
                .filter(age -> age <= 45)
```